relay-log/src/setup.rs

```diff
@@ -289,15 +295,31 @@ pub fn init(config: &LogConfig, sentry: &SentryConfig) {
     logs_subscriber.init();
 
     if let Some(dsn) = sentry.enabled_dsn() {
-        let guard = sentry::init(sentry::ClientOptions {
+        let mut options = sentry::ClientOptions {
             dsn: Some(dsn).cloned(),
             in_app_include: vec!["relay"],
             release: Some(RELEASE.into()),
             attach_stacktrace: config.enable_backtraces,
             environment: sentry.environment.clone(),
             traces_sample_rate: config.traces_sample_rate,
             ..Default::default()
-        });
+        };
+
+        // If `tags` is set in Sentry configuration install the before_send hook
+        // in order to inject said tags into each event
+        if let Some(tags) = sentry.tags.as_ref() {
+            // We need an object that will outlive `sentry` lifetime
+            let tags = tags.clone();
+            // Install hook
+            options.before_send = Some(Arc::new(move |mut event| {
+                event
+                    .tags
+                    .extend(tags.iter().map(|(k, v)| (k.to_owned(), v.to_owned())));
+                Some(event)
+            }));
+        }
```

Comment:
nit: You should be to simplify the code this way. You will clone every time but it should not be a big deal to clone `None`.

```suggestion
        if let Some(tags) = sentry.tags.clone() {
            // Install hook
            options.before_send = Some(Arc::new(move |mut event| {
                event
                    .tags
                    .extend(tags.iter().map(|(k, v)| (k.to_owned(), v.to_owned())));
                Some(event)
            }));
        }
```

Reply:
Can also replace `tags.iter().map(|(k, v)| (k.to_owned(), v.to_owned()))` with `tags.clone().into_iter()` :p